thinc/optimizers.py

```diff
@@ -100,7 +106,22 @@ def SGD(
     )
 
 
-class Optimizer(object):
+class OptimizerABC(Generic[GradT, WeightT]):
```

Comment:
Since we now have a `v9` branch where we can have larger changes, maybe it's good to discuss the naming and the class hierarchy a bit more internally?